### PR TITLE
Add OTLP helper to parse http request, and transform to SAPM

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

Split from https://github.com/signalfx/sapm-proto/pull/25
Depends on https://github.com/signalfx/sapm-proto/pull/28

### CODE REVIEW

File: otlp/constants.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+	tagMessage        = "message"
+	tagSpanKind       = "span.kind"
+	tagStatusCode     = "status.code"
+	tagStatusMsg      = "status.message"
+	tagError          = "error"
+	tagHTTPStatusCode = "http.status_code"
+
+	tagW3CTraceState          = "w3c.tracestate"
+	tagInstrumentationName    = "otlp.instrumentation.library.name"
```
Comment (tigrannajaryan) [inline otlp/constants.go:29]:
```suggestion
	tagInstrumentationName    = "otlp.instrumentation_library.name"
```

File: otlp/constants.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	tagInstrumentationVersion = "otlp.instrumentation.library.version"
```
Comment (tigrannajaryan) [inline otlp/constants.go:30]:
```suggestion
	tagInstrumentationVersion = "otlp.instrumentation_library.version"
```

Comment (tigrannajaryan) [reply otlp/constants.go:30]: Also possible change coming: https://github.com/open-telemetry/opentelemetry-specification/pull/989

File: otlp/parser.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"net/http"
+
+	splunksapm "github.com/signalfx/sapm-proto/gen"
+	otlpcoltrace "github.com/signalfx/sapm-proto/gen/otlp/collector/trace/v1"
+	"github.com/signalfx/sapm-proto/sapmprotocol"
+)
+
+// ParseRequest parses from the request (unzip if needed) an OTLP protobuf,
```
Comment (tigrannajaryan) [inline otlp/parser.go:25]:
```suggestion
// ParseRequest parses from the request (unzip if needed) from OTLP protobuf,
```

File: otlp/translator.go [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+
+	splunksapm "github.com/signalfx/sapm-proto/gen"
+	otlpcoltrace "github.com/signalfx/sapm-proto/gen/otlp/collector/trace/v1"
+	otlpcommon "github.com/signalfx/sapm-proto/gen/otlp/common/v1"
+	otlpresource "github.com/signalfx/sapm-proto/gen/otlp/resource/v1"
+	otlptrace "github.com/signalfx/sapm-proto/gen/otlp/trace/v1"
+)
+
+// otlpToSAPM translates otlp trace proto into the SAPM Proto.
+func otlpToSAPM(td otlpcoltrace.ExportTraceServiceRequest) (*splunksapm.PostSpansRequest, error) {
```
Comment (tigrannajaryan) [inline otlp/translator.go:31]: I assume this is copied from Collector. Does this need a review?

Comment (bogdandrutu) [reply otlp/translator.go:31]: The copied code is in https://github.com/signalfx/sapm-proto/pull/28

File: otlp/constants.go (hunk 2 above)
Comment (bogdandrutu) [reply otlp/constants.go:30]: Will fix in #28

File: otlp/constants.go (hunk 1 above)
Comment (bogdandrutu) [reply otlp/constants.go:29]: Will fix in #28

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**